test/integration/gssp-redirect-with-rewrites/test/index.test.js

```diff
@@ -0,0 +1,57 @@
+/* eslint-env jest */
+
+import { join } from 'path'
+import {
+  renderViaHTTP,
+  findPort,
+  launchApp,
+  killApp,
+  waitFor,
+} from 'next-test-utils'
+import webdriver from 'next-webdriver'
+
+// test suites
+
+const context = {}
+jest.setTimeout(1000 * 60 * 5)
+
+describe('getServerSideProps redirects', () => {
+  beforeAll(async () => {
+    context.appPort = await findPort()
+    context.server = await launchApp(join(__dirname, '../'), context.appPort, {
+      env: { __NEXT_TEST_WITH_DEVTOOL: 1 },
+    })
+
+    // pre-build all pages at the start
+    await Promise.all([
+      renderViaHTTP(context.appPort, '/alias-to-main-content'),
+      renderViaHTTP(context.appPort, '/main-content'),
+    ])
+  })
+  afterAll(() => killApp(context.server))
+
+  it('should use a client-side navigation for a rewritten URL', async () => {
```

Comment:
Let's add a similar test but instead ensure the `/unknown-route` being returned as a redirect does do a full navigation instead of a client navigation

Reply:
Done ! Thanks for your suggestions 😄